odoo/tools/misc.py

```diff
@@ -1530,3 +1532,13 @@ def traverse_containers(val, type_):
     elif isinstance(val, collections.abc.Sequence):
         for v in val:
             yield from traverse_containers(v, type_)
+
+
+def authenticate_message(env, message):
```

Comment:
Forwarding a few comments from @odony 
- a risk is that somebody "exposes" the helper in a public method and allows to generate free signatures (which would be bad). To mitigate, we can add a required parameter `scope` (`mass_mailing-mail_mail-open` for you for instance) to force people to indicate one (+docstring to explain).
- still on the helper name (sorry for the back and fourth), just `hmac` is clear enough and "grepable"
- no need to add the dbuuid, the combinaison of the dbsecret and the scope already serve the purpose of "uniqueness". Adding a fixed string does not add any security.
- hash method could be a parameter, defaulting a `sha256` but could allow others (for instance if need a short SHA1 hash)
- do not add a `sudo`, this way we force people to explicitly call the helper with sudo and recall them it's a bit sensible

Reply:
@mart-e The function name is ok ? (I just need to update the docstring for the point 2 ?)
(Because HMAC in already a function in the std lib, and dev might use the wrong function)

Reply:
@mart-e And the fonction can not be called in "SUDO" because it's just a function and not a model method

Reply:
But you can activate su mode on the environment. For the function name, the biggest risk is that dev do not use the helper (because they don't know it exists), if somebody greps "hmac" in the code, he will find our helper this way. We already have several tools method that have conflicting name like `add`, `round`, `encode`,.. but it's even sometimes a feature (`import safe_eval as eval`).

Reply:
@mart-e ok, let's call it hmac :)

about `env` I can not call sudo directly on it... Or I have to pass the secret but I prefer to use sudo in the function
![_tmp_screenshot](https://user-images.githubusercontent.com/54316226/81051334-d2853900-8ec1-11ea-935f-743311bdb908.png)

Reply:
My bad, it's possible with `env(su=True)` :)